pkg/ccl/backupccl/backup_test.go

```diff
@@ -3762,7 +3762,7 @@ func TestBackupRestoreDropDB(t *testing.T) {
 
 	sqlDB.Exec(t, "BACKUP DATABASE data TO $1", LocalFoo)
 	sqlDB.Exec(t, "CREATE DATABASE data2")
-	sqlDB.Exec(t, "RESTORE data.* FROM $1 WITH OPTIONS ('into_db'='data2')", LocalFoo)
```

Comment:
We shouldn't regress on any existing behavior -- it should be possible to handle 'into_db' = ... as well.

Reply:
I was under the impression that one of the expected consequences/improvements of moving away from the stringy map of KVOptions was that the user cannot specify a quoted option name. This would also have been the case after #50723 for all the BACKUP options, which would have previously accepted quotes in the name but now don't as they are hardcoded in the yacc grammar. This is also specified in that PR's description. If this is not the expected behavior then we should probably support all option names to be quoted/unquoted in both BACKUP and RESTORE?

Reply:
OK, as long as that was discussed and agreed on

Reply:
I'll allow @miretskiy to confirm once, I was largely making this change using the BACKUP switch as a reference.

Reply:
Yeah, we never documented that they were quotable strings -- all the examples in the docs are unquoted, so we said it was okay.